components/get_link_modal.tsx

```diff
@@ -1,104 +1,101 @@
-        } catch (err) {
-            this.setState({copiedLink: false});
-        }
-    }
+        if(textarea){
+            textarea.focus();
+            textarea.setSelectionRange(0, this.props.link.length);
 
-    render() {
-        let helpText = null;
-        if (this.props.helpText) {
-            helpText = (
-                <p>
-                    {this.props.helpText}
-                    <br/>
-                    <br/>
-                </p>
-            );
+            try {
+                this.setState({copiedLink: document.execCommand('copy')});
+            } catch (err) {
+                this.setState({copiedLink: false});
+            }
         }
+    }
 
-        let copyLink = null;
+    public render(): JSX.Element {
+        let helpText: JSX.Element = this.props.helpText ? (
+            <p>
+                {this.props.helpText}
+                <br/>
+                <br/>
+            </p>
+        ): <div/>;
 
-        if (document.queryCommandSupported('copy')) {
-            copyLink = (
-                <button
-                    id='linkModalCopyLink'
-                    data-copy-btn='true'
-                    type='button'
-                    className='btn btn-primary pull-left'
-                    onClick={this.copyLink}
-                >
-                    <FormattedMessage
-                        id='get_link.copy'
-                        defaultMessage='Copy Link'
-                    />
-                </button>
-            );
-        }
+        let copyLink: JSX.Element = document.queryCommandSupported('copy') ? (
+            <button
+                id='linkModalCopyLink'
+                data-copy-btn='true'
+                type='button'
+                className='btn btn-primary pull-left'
+                onClick={this.copyLink}
+            >
+                <FormattedMessage
+                    id='get_link.copy'
+                    defaultMessage='Copy Link'
+                />
+            </button>
+        ): <div/>;
 
-        const linkText = (
+        const linkText: JSX.Element = (
             <textarea
                 id='linkModalTextArea'
                 className='form-control no-resize min-height'
-                ref='textarea'
+                ref={this.textAreaRef}
                 value={this.props.link}
                 onClick={this.copyLink}
                 readOnly={true}
             />
         );
 
-        let copyLinkConfirm = null;
-        if (this.state.copiedLink) {
-            copyLinkConfirm = (
-                <p className='alert alert-success alert--confirm'>
-                    <SuccessIcon/>
-                    <FormattedMessage
-                        id='get_link.clipboard'
-                        defaultMessage=' Link copied'
-                    />
-                </p>
-            );
-        }
+        let copyLinkConfirm: JSX.Element = this.state.copiedLink ? (
```

Comment:
why the change in code? now this renders an empty div

Reply:
The code change was for the JSX.Element. Short-circuiting caused `copyLinkConfirm` to potentially be undefined which was not allowed with the JSX.Element.

However, based on your previous comment above JSX.Element type can be inferred so I can change this back to before the change.